vendor/k8s.io/kubernetes/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go

```diff
@@ -221,19 +220,18 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 				return false, nil
 			}
 
-			_, serverGroupsAndResources, err := crdClient.Discovery().ServerGroupsAndResources()
-			switch {
-			case clientdiscovery.IsGroupDiscoveryFailedError(err):
-				return false, nil
-			case err != nil:
-				return false, err
+			// The returned group and resource lists might be non-nil with partial results even in the
+			// case of non-nil error.  If API aggregation fails, we still want our other discovery information because the CRDs
+			// may all be present.
+			_, serverGroupsAndResources, discoveryErr := crdClient.Discovery().ServerGroupsAndResources()
+			if discoveryErr != nil {
+				glog.V(2).Info(discoveryErr)
```

Comment:
can we add some context there `glog.V(2).Infof("Failed to fetch discovery: %v", err)